Patch to prevent slowing down of zWay OpenHab binding

Wire82 opened the pull request:
Hi pathec,

during the last days I did some research to the OpenHab zWay binding because after the update to version 2.2 the zway binding slows down after some hours of use.
During the investigation of the problem, I found that starting and stopping of the Http Client within your API takes longer and longer over time and slows down the system.
The Patch shall prevent this by creating a new HttpClient every time a request is made. Testing this over several hours showed that the problem seems to be solved.
So please see this patch also as a starting point for further discussions.

Wire82 committed:
Fixed an issue that the HttpClient slows down after some time

Fixes:
Client is now established new every time before client is started
Update of class path
Update dependencies to latest versions


Signed-off-by: Johannes Einig <Johannes.Einig@me.com> (github: Wire82)

pathec commented:
Hi @Wire82,

thank you for your contribution. The changes look good. I'm testing the changes again. But due to the successful tests of the openHAB community, there should be no problems.

I'll get back to you. Have you already updated your Z-Way binding and created a PR in the openHAB repository? Otherwise, I'll do it.

pathec commented:
I have successfully tested the changes. 

Please squash the commits to one and update the Changelog. Then I approve the PR.

Wire82 commented:
Hi @pathec
I will do. Great to hear that the testing was successful. 
Until now I haven’t did a PR at OpenHAB. But I can do it now if the problem is solved. I also changed a few things within the binding to speed up some tasks.  
I will contact you if I have updated my fork. Than you can check the change and give maybe some feedback. 
If you agree with the changes I will make a PR to OpenHAB.

pathec commented:
Hi, @Wire82,
thanks for your support. Just let me know if you have any changes for the OpenAB Binding.

Wire82 commented:
Hi @pathec 
your are welcome. Today I prepared the PR for OpenHab. If you want you can check the changes I did for the binding. If every thing is correct I will make the PR within the next days.
The changes can be found here:
https://github.com/Wire82/openhab2-addons/tree/3106-ZWay-Binding-Patch
